include/shell.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <glib.h>
+
+#include "manifest.h"
+
+/**
+ * Add the meta-data contents from a manifest to a GPtrArray, formatted for use
+ * as shell variables.
```

Comment:
For clarity: Maybe note that newly allocated strings will be added to the GPtrArray?